libcaf_core/src/actor_system_config.cpp

```diff
@@ -125,6 +125,9 @@ actor_system_config::actor_system_config()
     .add<size_t>("max-consecutive-reads",
                  "max. number of consecutive reads per broker")
     .add<timespan>("heartbeat-interval", "interval of heartbeat messages")
```

Comment:
A bit of a conceptual weird point is that the `heartbeat-interval` of one node is related to the `connection-timeout` of its peers. Since heartbeats are uni-directional and don't provoke a response you have to be careful to keep that in mind. E.g., you can't have a configuration where some nodes are more interested in the liveliness status of their peers than others. Well you can, but the `heartbeat-interval` must fit the lowest timeout of all nodes.

I think we should make that more explicit in the description.

Reply:
Agreed. Ideally, BASP nodes would agree on a heartbeat interval during the handshake. However, for CAF 0.17.5, I'd like to avoid (breaking) changes to the network protocol. We should consider changing the handshake for 0.18, though.

Reply:
Hm, by the same reasoning: I think we should turn this off by default. If you are running CAF 0.17.4 and 0.17.5 side-by-side (e.g. during a migration phase), your 0.17.5 nodes otherwise sever a connection after 30s of no activity (0.17.4 has heartbeats *off* by default). We should recommend setting this for now and implement a better handshake with 0.18.